plugin/storage/factory.go

```diff
@@ -138,10 +138,15 @@ func (*Factory) getFactoryOfType(factoryType string) (storage.Factory, error) {
 func (f *Factory) Initialize(metricsFactory metrics.Factory, logger *zap.Logger) error {
 	f.metricsFactory = metricsFactory
 	for kind, factory := range f.factories {
+		role := "primary"
+		if _, ok := factory.(storage.ArchiveFactory); ok {
+			role = "archive"
+		}
```

Comment:
@yurishkuro I know this isn't correct but just put it here to start the conversation. This will currently set the role `archive` whenever the factory implements the ArchiveFactory. Is there a way to know at this point whether the storage is actually being used as `archive` since this is where the metrics factory is passed to the storage factory.

Reply:
Now that I think more about it, the v1 factory will initialize the CLI flags for both the primary namespace and the archive namespace. Since the factories today hold both the primary and archive configs, we probably need to do one of the following: 
- Each storage factory should only hold 1 config - This would be ideal but it would require creating two storage meta-factories which I think we decided we don't want. 
- Have the meta-factory pass all the metrics factories in with role=primary but override the role in `CreateArchiveSpanReader` to archive.

Let me know what you think

Reply:
Went with option 2 for now - if this looks good, we can make the same change for the other storage factories as well